Automattic/wp-calypso
# Automatically redirect to siteless checkout when logged-out

This PR updates the regular checkout flow to automatically redirect Jetpack users specifically coming from My Jetpack to the siteless checkout flow if they are not logged into WPCOM



See the related issue

## Proposed Changes

* This PR updates the checkout flow to redirect the user to a logged-out siteless checkout in case they are going through a regular checkout when coming from My Jetpack while logged-out on WordPress.com

| Logged-in      | Logged-out      |
| ------------- | ------------- |
| <img width="634" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5714212/0738c35e-b6b8-496a-a907-bdd3a338cf9e"> | <img width="633" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5714212/00bbfc73-01dd-494e-a6c0-e78f9aaeff66"> |


## Why are these changes being made?


* We are adding this change so users coming from My Jetpack will be redirected to the siteless checkout in case they are not logged into WPCOM. This will continue working as before for users who visit that checkout flow while logged-in.

## Testing Instructions



* Open the Calypso live link in incognito
* Make sure to have a test site that is user-connected
* Copy this path (you can also confirm that a similar checkout URL is generated by any interstitial in My Jetpack of your test site) `/checkout/{site_id}/jetpack_backup_t1_yearly?site={site_slug}&source=my-jetpack&redirect_to=http%3A%2F%2F{site_slug}%2Fwp-admin%2Fadmin.php%3Fpage%3Dmy-jetpack`
* Be sure to replace site_id and site_slug with your site's information
* In the incognito tab with the Calypso live page, paste what you've just copied after the Calypso live domain
* Confirm that you are redirected to the siteless checkout instead of the regular checkout
* If you copy and paste the same path (in step 3) to WordPress.com in incognito, you'd be redirected to a page prompting for login
* If you open the same URL mentioned above while logged-in, you'll be redirected to a "sited checkout"

## Pre-merge Checklist
